docs/DeveloperGuide.md

```diff
@@ -36,3 +33,470 @@
+
+#### ExecuteShoot Method
+
+The `executeShoot` method exemplifies polymorphism in action, adjusting player behavior during shooting based on their skill level.
+
+```java
+public static void executeShoot(String[] readArgumentTokens) {
+  float adjustedDirection = PlayerList.l1.get(Ui.curPlayer).shootDirectionAdjust(selectedDirectionIndex);
+  float adjustedAiDirection = PlayerList.l1.get(Ui.curPlayer).aiDirectionAdjust(Ai.getAiDirection());
+  float adjustedRange = PlayerList.l1.get(Ui.curPlayer).rangeAdjust();
+  //...
+  PlayerList.l1.get(Ui.curPlayer).printGoalAfterShoot(isScoreGoal, Math.round(adjustedDirection));
+}
+```
+### Skill Upgrade Mechanism
+
+The `skillUpgrade` method in the `PlayerList` class is a key feature, allowing players to improve their skills based on game performance. The method checks the player's current skill and match count, upgrading their skill level if certain conditions are met.
+he mechanism operates through a structured process, enhancing the gaming experience by offering a realistic approach to skill progression:
+
+1. **Skill Level Assessment**: The system evaluates a player's current skill level alongside their performance metrics, such as match count and success rate.
+
+2. **Determination of Skill Upgrade Eligibility**: Utilizing predefined criteria, the system determines if a player qualifies for a skill upgrade. These criteria may include metrics like a minimum number of matches played or specific performance thresholds.
+
+3. **Application of Skill Upgrade**: Eligible players will have their skill attribute adjusted to a higher level. This upgrade might unlock new abilities, enhance stats, or offer other in-game advantages.
+
+4. **Feedback to the User**: The system informs the user of the successful skill upgrade through UI messages or other feedback mechanisms, providing immediate recognition of their achievement.
+
+![UML Class Diagram](diagrams%2FupdateSkill.png)
+
+### Alternatives Considered
+
+#### Alternative Collections for Player Management
+- **HashSet<Player>**: Evaluated for its uniqueness properties and constant-time performance for add, remove, and contains operations. It was ultimately not selected due to the ordered nature of player management required in our game, which is better served by a List implementation.
+
+#### Polymorphism Implementation Methods
+- **Interface-based Polymorphism**: An alternative approach using interfaces to define common behaviors for all players was considered. While this would have provided a clean separation of concerns, the decision to use class inheritance and method overriding was driven by the shared attributes and methods across player types, favoring a more hierarchical structure.
```

Comment:
I appreciate the developers outlining the alternatives considered, and the rationale for choosing current implementation instead of alternatives. For example, outlining the rationale of using List instead of HashSet due to the ordered nature of data aids my visualisation of how the data is structured